### PR TITLE
[FEA] PyTorch Dataloader support for multi-GPU training w/ Horovod

### PR DESCRIPTION
**Pull request by karlhigley**
This is a working training script for the Movielens example with multiple GPUs but no data partitioning. (All workers train on exactly the same data, since the dataloader itself doesn't yet have support for partitioning the data across workers.)

### CODE REVIEW

**Comment by jperez999** [comment]
Can we add the pytorch script to the examples/horovod or move the tf script and the hvd_wrapper.sh file to the appropriate directory (i.e. examples/multi-gpu...)?

**Review by jperez999 — CHANGES_REQUESTED** [review]

File: nvtabular/loader/backend.py [hunk 1]
```diff
@@ -173,13 +173,14 @@ def __init__(
         label_names,
         batch_size,
         shuffle,
+        seed_fn=None,
```
[the remark this answers is not in the document]
Comment (karlhigley) [reply nvtabular/loader/backend.py:176]: Yeah, happy to open another PR to add it to the TF side.

File: nvtabular/loader/backend.py [hunk 2]
```diff
@@ -173,13 +173,14 @@ def __init__(
...
         parts_per_chunk=1,
         devices=None,
         global_size=None,
         global_rank=None,
     ):
         self.data = dataset
-        self.indices = cp.arange(dataset.to_ddf().npartitions)
+        self.indices = indices or cp.arange(dataset.to_ddf().npartitions)
```
[the remark this answers is not in the document]
Comment (karlhigley) [reply nvtabular/loader/backend.py:183]: Good catch, missed this in the merge conflict resolution.

File: tests/unit/test_torch_dataloader.py [hunk 3]
```diff
@@ -318,3 +320,79 @@ def rmspe_func(y_pred, y):
...
+
+    process = subprocess.Popen(
+        [
+            "horovodrun",
+            "-np",
+            "2",
+            "-H",
+            "localhost:2",
+            "python",
+            "examples/multi-gpu/torch-nvt-horovod.py",
```
[the remark this answers is not in the document]
Comment (karlhigley) [reply tests/unit/test_torch_dataloader.py:386]: Yeah, we definitely should. I'll borrow it from your PR.

**Review by benfred — APPROVED** [review]
looks great!

**Review by jperez999 — APPROVED** [review]

**Review by benfred — APPROVED** [review]

**Pull request merged**